### ISSUE TITLE
Improved TypeScript types for `style`

### ISSUE THREAD

**Issue by zaydek** [facebook/react-strict-dom#61]
### Describe the feature request

tldr: _Please make it possible to compose `style` using TypeScript._

Hi there, I noticed that when trying to compose types such as the following:

<img width="411" alt="Screenshot 2024-03-24 at 6 23 19 PM" src="https://github.com/facebook/react-strict-dom/assets/58870766/3c6179dd-e5be-4469-a3a3-bd989a2b4857">

`props.style` only satisfies TypeScript if I do `props.style ?? false`. `style={[styles.modal, false]}` is accepted but `style={[styles.modal, undefined is not]}`, which is why `?? false` works.

The other example is `style={[styles.modal, { backgroundColor }]}` which I belive is intended to be supported by looking at the `apps/examples`.

Here are the complete errors:

```
Type 'Styles | undefined' is not assignable to type 'false | unique symbol | Readonly<Readonly<{ readonly theme?: StyleXClassNameFor<"theme", Readonly<string | null | undefined>> | undefined; readonly MozOsxFontSmoothing?: StyleXClassNameFor<...> | undefined; ... 542 more ...; readonly '::-webkit-search-results-decoration'?: StyleXClassNameFor<...> | undefined; }> & P...'.
  Type 'undefined' is not assignable to type 'false | unique symbol | Readonly<Readonly<{ readonly theme?: StyleXClassNameFor<"theme", Readonly<string | null | undefined>> | undefined; readonly MozOsxFontSmoothing?: StyleXClassNameFor<...> | undefined; ... 542 more ...; readonly '::-webkit-search-results-decoration'?: StyleXClassNameFor<...> | undefined; }> & P...'.ts(2322)
(parameter) props: ModalProps
```

```
Type '{ backgroundColor: string; }' is not assignable to type 'false | unique symbol | Readonly<Readonly<{ readonly theme?: StyleXClassNameFor<"theme", Readonly<string | null | undefined>> | undefined; readonly MozOsxFontSmoothing?: StyleXClassNameFor<...> | undefined; ... 542 more ...; readonly '::-webkit-search-results-decoration'?: StyleXClassNameFor<...> | undefined; }> & P...'.
  Types of property 'backgroundColor' are incompatible.
    Type 'string' is not assignable to type 'StyleXClassNameFor<"backgroundColor", Readonly<string | null | undefined>>'.
      Type 'string' is not assignable to type '{ _opaque: unique symbol; _key: "backgroundColor"; _value: Readonly<string | null | undefined>; }'.ts(2322)
(property) backgroundColor?: StyleXClassNameFor<"backgroundColor", Readonly<string | null | undefined>> | undefined
```

Am I doing something wrong or are the TypeScript types slightly off?

Here's the example I was referring to, so it looks like both of these are expected to work if not for the type errors: https://github.com/facebook/react-strict-dom/blob/main/apps/examples/src/App.js#L216.

**Comment by necolas** [comment]
Inline styles are not allowed in RSD / StyleX, there is a function syntax for that. The example in this repo is using an out of date style. 

The issue with `undefined` might be coming from the Flow translation to TS. cc @nmn

**Comment by nmn** [comment]
Inline styles are intentionally not supported because of specificity issues when merging styles on the web. (Inline styles are always more specific than CSS).

We are considering adding an API to StyleX on the lines of `stylex.inline()` that would allow something similar.

To debug the `undefined` problem, I would need a bit more context. Please open an issue on the StyleX repo with a repro.

**Comment by nmn** [comment]
There is now a StyleX issue with a proposal for `stylex.inline()`. That is now the correct place for this discussion:

https://github.com/facebook/stylex/issues/534

**Comment by necolas** [comment]
Isn't there still the `undefined` issue with the TS types?

> style={[styles.modal, false]} is accepted but style={[styles.modal, undefined is not]}

**Comment by nmn** [comment]
> Isn't there still the undefined issue with the TS types?

Yeah, looks like an edge-case bug in `flow-api-translator` that causes `?false` from Flow to skip `undefined` and generate `null | false`.

The source of this bug is within StyleX's type defs, so I'll fix it there.
